client/data/courses/types.ts

```diff
@@ -0,0 +1,24 @@
+export type CourseSlug = 'blogging-quick-start';
+
+export type VideoSlug = string;
+
+export interface CourseVideo {
+	title: string;
+	description: string;
+	duration_seconds: string;
+	poster: string;
+	url: string;
+	completed_seconds: string;
+}
+
+export interface Course {
+	title: string;
+	slug: string;
+	cta: {
+		description: string;
+		action: string;
+		url: string;
+	};
+	videos: CourseVideo[];
+	completions: { [ key in VideoSlug ]: boolean | boolean[] };
```

Comment:
The returned data might be a boolean array if you completed the video multiple times. But it might be the edge case because it would only complete the video if the video is never completed.